src/layer/transportation_label.js

```diff
@@ -0,0 +1,141 @@
+const motorwayToTrunk = ["motorway", "trunk"];
+const motorwayToPrimary = [...motorwayToTrunk, "primary"];
+const motorwayToSecondary = [...motorwayToPrimary, "secondary"];
+const motorwayToMinor = [...motorwayToSecondary, "tertiary", "minor"];
+const motorwayToService = [...motorwayToMinor, "service"];
+
+const majorConstruction = ["motorway_construction", "trunk_construction"];
+const minorConstruction = [
+  "primary_construction",
+  "secondary_construction",
+  "tertiary_construction",
+  "minor_construction",
+  "service_construction",
+];
+
+export const label = {
+  id: "road_label",
+  type: "symbol",
+  paint: {
+    "text-color": [
+      ...highwaySelector,
+      majorConstruction,
+      "maroon",
+      minorConstruction,
+      "slategray",
+      "ferry",
+      "hsl(211, 53%, 15%)",
+      "#333",
+    ],
+    "text-halo-color": [
+      ...highwaySelector,
+      "ferry",
+      "hsl(211, 70%, 90%)",
+      Color.backgroundFill,
+    ],
+    "text-halo-blur": 0.5,
+    "text-halo-width": 2,
+    "text-opacity": [
+      "step",
+      ["zoom"],
+      [...highwaySelector, "motorway", 1, 0],
+      10,
+      [...highwaySelector, motorwayToTrunk, 1, 0],
+      11,
+      [...highwaySelector, motorwayToPrimary, 1, "ferry", 1, 0],
+      12,
+      [...highwaySelector, motorwayToSecondary, 1, "ferry", 1, 0],
+      13,
+      [...highwaySelector, motorwayToMinor, 1, "ferry", 1, 0],
+      14,
+      [...highwaySelector, motorwayToService, 1, "ferry", 1, 0],
+      15,
+      1,
+    ],
+  },
+  filter: [
+    "in",
+    ["get", "class"],
+    [
+      "literal",
+      [
+        ...motorwayToService,
+        ...majorConstruction,
+        ...minorConstruction,
+        "ferry",
+      ],
+    ],
+  ],
+  layout: {
+    "text-font": [
+      ...highwaySelector,
+      "ferry",
+      ["literal", ["Metropolis Regular Italic"]],
+      ["literal", ["Metropolis Light"]],
+    ],
+    "text-field": "{name:latin} {name:nonlatin}",
+    "text-max-angle": 20,
+    "symbol-placement": "line",
+    "text-size": [
+      "step",
+      ["zoom"],
+      12,
+      16,
+      [...highwaySelector, motorwayToTrunk, 10, 12],
+      17,
+      [...highwaySelector, motorwayToSecondary, 10, 12],
+    ],
+    "text-anchor": [
+      "step",
+      ["zoom"],
+      "bottom",
+      16,
+      [...highwaySelector, motorwayToTrunk, "center", "bottom"],
+      17,
+      [...highwaySelector, motorwayToSecondary, "center", "bottom"],
+    ],
+    "symbol-sort-key": [
+      // TODO busway
+      ...highwaySelector,
+      "motorway",
+      0,
+      "trunk",
+      1,
+      ["primary", "ferry"],
+      2,
+      "secondary",
+      3,
+      ["tertiary", "minor"],
+      4,
+      5,
+    ],
+  },
+  source: "openmaptiles",
+  "source-layer": "transportation_name",
+};
+
+// A spacer label on each bridge to push any waterway label away from the bridge.
+// https://github.com/ZeLonewolf/openstreetmap-americana/issues/198
+export const bridgeSpacer = {
```

Comment:
I forgot about this – we might want to keep waterway labels away from railroad bridges as well in #476.

Reply:
I suspect we already are:

![image](https://user-images.githubusercontent.com/23022/181485897-fae6b7ca-e774-4942-aac4-fbde4c61fd21.png)

Reply:
Or at least, this layer is not filtering any class out of `transportation`. Unclear if there's a bug:

![image](https://user-images.githubusercontent.com/23022/181486474-5fe07b05-9845-4104-a300-b038a86ecbb8.png)